runtime/lua/vim/lsp/_watchfiles.lua

```diff
@@ -1,152 +1,80 @@
 local bit = require('bit')
+local lpeg = require('lpeg')
 local watch = require('vim._watch')
 local protocol = require('vim.lsp.protocol')
 
 local M = {}
 
 ---@private
----Parses the raw pattern into a number of Lua-native patterns.
+---Parses the raw pattern into an |lpeg| pattern.
```

Comment:
worth a brief comment explaining the motivation:

> since LPeg natively supports alternatives (match "this" or "that"), then we can output a single LPeg pattern from a glob instead of a number of Lua patterns

Reply:
Added